examples/platform/linux/AppMain.cpp

```diff
@@ -490,10 +490,10 @@ class ExampleDeviceInstanceInfoProvider : public DeviceInstanceInfoProvider
     CHIP_ERROR GetHardwareVersionString(char * buf, size_t bufSize) override
     {
         // Check if it was set from the command line or fall back to default provider.
-        if (mpHardwareVersionString)
+        if (mHardwareVersionString.HasValue())
         {
-            VerifyOrReturnError(bufSize > strlen(mpHardwareVersionString), CHIP_ERROR_BUFFER_TOO_SMALL);
-            strcpy(buf, mpHardwareVersionString);
+            VerifyOrReturnError(CanFitInNullTerminatedString(mHardwareVersionString.Value(), bufSize), CHIP_ERROR_BUFFER_TOO_SMALL);
+            strcpy(buf, mHardwareVersionString.Value().c_str());
```

Comment:
Don't use `strcpy` directly, use `CopyString(buf, bufSize, mHardwareVersionString.Value().c_str())` and similar, everywhere, from `<lib/support/CHIPMemString.h>`. Never use `strcpy` directly even after having checked the bounds.